src/Microsoft.DotNet.XHarness.iOS.Shared/Utilities/Helpers.cs

```diff
@@ -60,6 +62,19 @@ public Guid GenerateStableGuid(string seed = null)
 
         public string GetTerminalName(int filedescriptor) => Marshal.PtrToStringAuto(ttyname(filedescriptor));
 
-        public IEnumerable<IPAddress> GetLocalIpAddresses() => Dns.GetHostEntry(Dns.GetHostName()).AddressList;
+        public IEnumerable<IPAddress> GetLocalIpAddresses()
+        {
+            var addresses = new List<IPAddress>();
+
+            foreach (var networkInterface in NetworkInterface.GetAllNetworkInterfaces().Where(adapter => adapter.OperationalStatus == OperationalStatus.Up))
+            {
+                foreach (UnicastIPAddressInformation info in networkInterface.GetIPProperties().UnicastAddresses)
+                {
+                    addresses.Add(info.Address);
+                }
+            }
+
+            return addresses;
```

Comment:
I'd filter out loopback addresses and also put IPv4 ones first in the list, since these are the more common ones:

```suggestion
            var ipv4Addresses = new List<IPAddress>();
            var otherAddresses = new List<IPAddress>();

            foreach (var networkInterface in NetworkInterface.GetAllNetworkInterfaces().Where(adapter => adapter.OperationalStatus == OperationalStatus.Up))
            {
                foreach (UnicastIPAddressInformation info in networkInterface.GetIPProperties().UnicastAddresses.Where(info => !IPAddress.IsLoopback(info.Address)))
                {
                    if (info.Address.AddressFamily == AddressFamily.InterNetwork)
                    {
                        ipv4Addresses.Add(info.Address);
                    }
                    else
                    {
                        otherAddresses.Add(info.Address);
                    }
                }
            }

            return ipv4Addresses.Concat(otherAddresses);
```

Reply:
Nice, this makes sense a lot!

Reply:
This was the result in Helix:
```
10.105.153.238
169.254.71.106
fe80::1%1
fe80::c42:834b:569e:4bfe%4
2001:4898:9008:60:8c7:1432:bc6e:ed0
2001:4898:9008:60:25b1:d447:cdfe:7e94
2001:4898:9008:60:9de5:3e76:e45b:3067
2001:4898:9008:60:543a:6fa5:a658:83f5
2001:4898:9008:60:80bf:4058:bb60:91f8
2001:4898:9008:60:599f:90fd:50d:9916
2001:4898:9008:60:ecba:fa9e:abb:b4ca
2001:4898:9008:60:249f:a1b5:1fd5:4da3
fe80::aede:48ff:fe00:1122%5
fe80::1:7c3b:1520:da73%6
fe80::347d:c3ff:fe9b:f4%10
fe80::347d:c3ff:fe9b:f4%11
fe80::c387:2f7d:127a:582a%17
fe80::483:9ad:2efa:2866%18
```

LGTM!